### ISSUE TITLE
New option for padded-blocks

### ISSUE THREAD

**Issue by rightaway** [eslint/eslint#4414]
`padded-blocks` has 2 options, `always` and `never`. I suggest a 3rd one, for example `whenBlankLines`.

The idea is that if the block has any blank lines within it, then it should act like `always`. If it doesn't have any blank lines, then it should act like `never`.

Here are examples of code that wouldn't product warnings

```js
// Acts like "never" because no blank lines between one() and three()
function a() {
  one();
  two();
  three();
}

// Acts like "always" because there's blank lines, like between the if block and two()
function b() {

  if (true) {
    one();
  }

  if (true) {
    two();
  }
  
  three();

}
```

This in my opinion looks nicer than

```js
function b() {
  if (true) {
    one();
  }

  if (true) {
    two();
  }
  
  three();
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
So what you are asking for is if there is at least one blank line anywhere within a given block, block should be padded, right?

**Comment by rightaway** [comment]
@ilyavolodin That's right.

**Comment by ilyavolodin** [comment]
I guess it's reasonable option. @eslint/eslint-team thoughts?

**Comment by IanVS** [comment]
Hmmm. I understand how the examples given might arguably look better that way, but it still feels to me like introducing inconsistencies which the whole point of using ESLint is to avoid them.  Besides, switching behavior based on finding a single blank line seems like a weak heuristic.  I would argue to either pick one style or the other, or turn the rule off.  Or of course it could be a custom rule.  Overall, I'm :-1: on adding to core, for what it's worth.

**Comment by rightaway** [comment]
@IanVS Something doesn't have to be identical to be consistent. For the same reason many of the options for http://eslint.org/docs/rules/curly allow for different variations that follow the same pattern, wheree for example if there's one statement, don't require curly braces, but if there's more than one, then require them. It's still being consistent, allowing for the best variation of the code based on what looks best for that situation. Plus it's not a breaking change, just an option for people who agree as you said that it looks better that way.

**Comment by nzakas** [comment]
Yeah, I feel like this is a pretty fuzzy heuristic that I don't see getting a lot of use. It would involve a bunch of backtracking that I think would overcomplicate the rule. I'd suggest creating a custom rule if you want this functionality.
